pkg/build/api/validation/validation.go

```diff
@@ -213,13 +205,33 @@ func validateSourceStrategy(strategy *buildapi.SourceBuildStrategy) fielderrors.
 
 func validateCustomStrategy(strategy *buildapi.CustomBuildStrategy) fielderrors.ValidationErrorList {
 	allErrs := fielderrors.ValidationErrorList{}
-	if strategy.From == nil || len(strategy.From.Name) == 0 {
+	if strategy.From == nil {
 		allErrs = append(allErrs, fielderrors.NewFieldRequired("from"))
+	} else {
+		allErrs = append(allErrs, validateStrategyFrom(strategy.From).Prefix("from")...)
 	}
 	allErrs = append(allErrs, validateSecretRef(strategy.PullSecret).Prefix("pullSecret")...)
 	return allErrs
 }
 
+func validateStrategyFrom(from *kapi.ObjectReference) fielderrors.ValidationErrorList {
+	allErrs := fielderrors.ValidationErrorList{}
+	if from.Kind != "ImageStreamImage" && from.Kind != "ImageStreamTag" && from.Kind != "DockerImage" {
```

Comment:
Actually from what I see in the defaulting funcs it means ImageStreamTag for from (see [here](https://github.com/openshift/origin/blob/master/pkg/build/api/v1beta3/conversion.go#L16-L30)). Should it be IS?